src/units/abilities.cpp

```diff
@@ -1113,6 +1113,64 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 	}
 }
 
+unit_ability_list attack_type::get_specials_and_abilities_with_priority(const std::string& special, std::vector<std::string> checking_string) const
+{
+	//the custom_priority attribute is used for define rules of priority of damage type
+	unit_ability_list abil_list = get_specials_and_abilities(special);
+	unit_ability_list temp_list;
+	if(abil_list.empty()){
+		return abil_list;
+	}
+
+	std::vector<double> priorityvec;
+	std::vector<double>::iterator priority;
+	for(const auto& i : abil_list) {
+		priorityvec.push_back((*i.ability_cfg)["priority"].to_double(0));
+	}
+
+
+	priority = std::max_element(priorityvec.begin(), priorityvec.end());
+	int prc = std::count(priorityvec.begin(), priorityvec.end(), *priority);
```

Comment:
A couple of lines here are using exact equality with floating-point values. I haven't yet worked out what it should do instead, just making a note for myself.

Reply:
It's not _necessarily_ incorrect to use exact equality with floating-point values. It might be fine in this context – we're not doing math on the values, just parsing them from text and comparing them to each other.